ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB/Model.hs

```diff
@@ -121,6 +124,14 @@ getBlockModel err sl = do
     if not open then EH.throwError' err $ UserError ClosedDBError
     else return $ Map.lookup sl mp
 
+getHeaderModel :: forall m blockId. (MonadState (DBModel blockId) m, Ord blockId)
+               => ThrowCantCatch VolatileDBError m
+               -> blockId
+               -> m (Maybe (SlotNo, ByteString))
+getHeaderModel err sl = do
+    bs <- getBlockModel err sl
+    return $ second (BL.take 1 . BL.drop (fromIntegral headerOffset)) <$> bs
```

Comment:
Can you introduce `headerSize = 1` in `VolatileDB.TestBlock`? Otherwise this is just undocumented magic.

I think the following would be clearer:

```haskell
getHeaderModel err sl = second extractHeader <$> getBlockmodel err sl
  where
    extractHeader = BL.take (fromIntegral headerSize)
                  . BL.drop (fromIntegral headerOffset)
```